velox/expression/Expr.cpp

```diff
@@ -200,13 +210,23 @@ void Expr::computeMetadata() {
     }
     deterministic_ &= input->deterministic_;
     if (!input->distinctFields_.empty()) {
-      propagatesNulls_ &= input->propagatesNulls_;
+      if (input->propagatesNulls_) {
```

Comment:
This logic is needed only when isNullPropagatingFunction is true. Should we skip this logic otherwise?